packages/client/hmi-client/src/workflow/ops/stratify-mira/tera-stratify-mira.vue

```diff
@@ -153,98 +151,66 @@ const stratifyModel = () => {
-		msgType: 'reset_request',
-		msgId: createMessageId('stratify_request')
-	};
-	const contextMessage: JupyterMessage = createMessage(messageBody);
-	kernel.sendJupyterMessage(contextMessage);
+	if (!model.value) return;
+
+	const resetMessage = kernelManager.sendMessage('reset_request', {});
+
+	if (resetMessage) {
+		resetMessage.register('reset_response', handleResetResponse);
+	}
+};
+
+const handleResetResponse = (data: any) => {
+	console.log(data.content);
 };
 
 const stratifyRequest = () => {
-	const kernel = jupyterSession.value?.session?.kernel;
-	if (!kernel || !model.value) return;
+	if (!model.value) return;
 
 	const strataOption = props.node.state.strataGroups[0];
-	const messageBody = {
-		session: jupyterSession?.value?.name || '',
-		channel: 'shell',
-		content: {
-			stratify_args: {
-				key: strataOption.name,
-				strata: strataOption.groupLabels.split(',').map((d) => d.trim()),
-				concepts_to_stratify: strataOption.selectedVariables,
-				cartesian_control: strataOption.cartesianProduct
-			}
-		},
-		msgType: 'stratify_request',
-		msgId: createMessageId('stratify_request')
+	const messageContent = {
+		stratify_args: {
+			key: strataOption.name,
+			strata: strataOption.groupLabels.split(',').map((d) => d.trim()),
+			concepts_to_stratify: strataOption.selectedVariables,
+			cartesian_control: strataOption.cartesianProduct
+		}
 	};
-	const contextMessage: JupyterMessage = createMessage(messageBody);
-	kernel.sendJupyterMessage(contextMessage);
-};
-
-const setKernelContext = () => {
-	const kernel = jupyterSession.value?.session?.kernel;
-	if (!kernel || !model.value) {
-		return;
-	}
 
-	const messageBody = {
-		session: jupyterSession?.value?.name || '',
-		channel: 'shell',
-		content: {
-			context: 'mira_model',
-			language: 'python3',
-			context_info: {
-				id: model.value.id
-			}
-		},
-		msgType: 'context_setup_request',
-		msgId: createMessageId('context_setup')
-	};
-	const contextMessage: JupyterMessage = createMessage(messageBody as any);
-	kernel.sendJupyterMessage(contextMessage);
-};
+	const stratifyMessage = kernelManager.sendMessage('stratify_request', messageContent);
 
-const iopubMessageHandler = (_session: any, message: any) => {
-	if (message.header.msg_type === 'status') {
-		return;
+	if (stratifyMessage) {
+		stratifyMessage
+			.register('stratify_response', handleStratifyResponse)
+			.register('model_preview', handleModelPreview);
 	}
+};
 
-	const msgType = message.header.msg_type;
-
-	if (msgType === 'stratify_response') {
-		const codes = message.content.executed_code.split('\n');
-		codes.forEach((c: any) => {
-			console.log(c);
-		});
-	} else if (msgType === 'reset_response') {
-		console.log(message.content);
-	} else if (msgType === 'model_preview') {
-		model.value = message.content['application/json'];
-	}
+const handleStratifyResponse = (data: any) => {
+	const codes = data.content.executed_code.split('\n');
+	codes.forEach((c: any) => {
+		console.log(c);
+	});
```

Comment:
If this is for debugging, it needs to go.

Reply:
We will need this for dev - the task is only 50% complete.